src/network/tcp.py

```diff
@@ -67,18 +68,19 @@ def __init__(self, address=None, sock=None):
             self.connect(self.destination)
             logger.debug("Connecting to %s:%i", self.destination.host, self.destination.port)
         encodedAddr = protocol.encodeHost(self.destination.host)
-        if protocol.checkIPAddress(encodedAddr, True) and not protocol.checkSocksIP(self.destination.host):
-            self.local = True
-        else:
-            self.local = False
-        #shared.connectedHostsList[self.destination] = 0
-        ObjectTracker.__init__(self)
+        self.local = all([
+            protocol.checkIPAddress(encodedAddr, True),
+            not protocol.checkSocksIP(self.destination.host)
+        ])
+        ObjectTracker.__init__(self)  # pylint: disable=non-parent-init-called
         self.bm_proto_reset()
         self.set_state("bm_header", expectBytes=protocol.Header.size)
 
-    def antiIntersectionDelay(self, initial = False):
+    def antiIntersectionDelay(self, initial=False):
+        """TBC"""
```

Comment:
This is a defense against the so called intersection attacks. It is called when you notice peer is requesting non-existing objects, or right after the connection is established. It will estimate how long an object will take to propagate across the network, and skip processing "getdata" requests until then. This means an attacker only has one shot per IP to perform the attack.